### DISCUSSION TITLE
fatal error C1083: Cannot open include file: 'any'...  -- How to find MSVC 19.0.24241.7

### DISCUSSION THREAD

**Discussion by russellsquared** [nlohmann/json#4119] [Q&A]
I am stuck using VS2015. I have Update 3 installed, as well as vs14-kb4567703 and vc14-kb4020481. That gets me to `
Microsoft (R) Build Engine version 14.0.25420.1
The C compiler identification is MSVC 19.0.24223.0`

I cannot find anywhere on the internet how to get to `MSVC 19.0.24241.7` as referenced on the supported compilers. Do you have a link to an installer or package to get to that version of MSVC? Even opening VS2015 IDE and looking for updates, I don't find anything relating to that version.

Thanks for the guidance

**Comment by nlohmann** [comment]
This is a tooling/environment question rather than a library question: <any> is a C++17 standard header, and VS2015 (MSVC 19.0.x) predates full C++17 support, so it doesn't ship <any> at all regardless of update level — there's no specific MSVC 19.0.24241.7 package that adds it. The specific compiler-version numbers in the library's supported-compiler table are just what was tested, not a hard requirement; practically you'd need to upgrade to a newer Visual Studio (2017+) that ships <any> and better C++17 support, since VS2015 cannot satisfy the header dependency no matter which update is installed.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
